frontend/src/stores/test-suite.ts

```diff
@@ -0,0 +1,114 @@
+import {
+    DatasetDTO,
+    JobDTO,
+    ModelDTO,
+    TestCatalogDTO,
+    TestSuiteExecutionDTO,
+    TestSuiteNewDTO
+} from '@/generated-sources';
+import {defineStore} from 'pinia';
+import {api} from '@/api';
+import {chain} from 'lodash';
+import {trackJob} from '@/utils/job-utils';
+import {useMainStore} from '@/stores/main';
+
+interface State {
+    projectId: number | null,
+    inputs: { [name: string]: string },
+    suite: TestSuiteNewDTO | null,
+    registry: TestCatalogDTO | null,
+    datasets: { [key: string]: DatasetDTO },
+    models: { [key: string]: ModelDTO },
+    executions: TestSuiteExecutionDTO[],
+    trackedJobs: { [uuid: string]: JobDTO }
+}
+
+const mainStore = useMainStore();
+
+export const useTestSuiteStore = defineStore('testSuite', {
+    state: (): State => ({
+        projectId: null,
+        inputs: {},
+        suite: null,
+        registry: null,
+        datasets: {},
+        models: {},
+        executions: [],
+        trackedJobs: {}
+    }),
+    getters: {
+        suiteId: ({suite}) => suite === null ? null : suite.id,
+        testSuiteResults: ({executions}) => {
+            if (!executions) {
+                return {};
+            }
+
+            return chain(executions)
+                .map(execution => (execution.results ?? []).map(
+                    result => ({
+                        testResult: result,
+                        testSuiteResult: execution
+                    })
+                ))
+                .flatten()
+                .groupBy(result => result.testResult.test.testId)
+                .value();
+        }
+    },
+    actions: {
+        async reload() {
+            if (this.suiteId !== null && this.projectId !== null) {
+                await this.loadTestSuite(this.projectId, this.suiteId)
+            }
+        },
+        async loadTestSuite(projectId: number, suiteId: number) {
+            const [
+                inputs,
+                suite,
+                registry,
+                datasets,
+                models,
+                executions
+            ] = await Promise.all([
+                api.getTestSuiteNewInputs(projectId, suiteId),
```

Comment:
six requests is a bit too much, let's create a custom DTO and send all of the required info in one go